### PR TITLE
Basic authentication testing

### PR DESCRIPTION
**Pull request by RogerTangos**
- write functional tests making sure that the login NOT required pages all show up, and don't have broken links
- write functional tests for creating logging into an account (manually)
- add id tags to parts of the layout, so that the tests work
- update the docs a bit 
- fix some links that the docs create

### CODE REVIEW

**Comment by RogerTangos** [comment]
I'm closing this and re-opening with some changes I just made.

**Comment by RogerTangos** [comment]
Hey Justin - can you take a look at this one?

File: src/www/templates/index.html [hunk 1]
```diff
@@ -46,7 +46,7 @@ <h1 id="overview" class="page-header">What is DataHub?</h1>
         <li>A database-agnostic, language-agnostic, <a href="https://github.com/mitreid-connect/">MITREid (OpenID + OAuth2)</a> integrated platform for your mobile and web apps</li>
         <li>A web client for easy manipulation of your data</li>
         <li>An application ecosystem for data processing, including ingestion, curation, integration, discovery, query, analytics, visualization, and machine learning</li>
-        <li>An open source project from <a href="https://csail.mit.edu">MIT CSAIL's</a> <a href="livinglab.mit.edu">Living Lab</a></li>
+        <li>An open source project from <a href="csail.mit.edu">MIT CSAIL's</a> <a href="livinglab.mit.edu">Living Lab</a></li>
```
Comment (justinanderson) [inline src/www/templates/index.html]: These hrefs need to start with a scheme, or they'll be interpreted as relative paths.

File: src/functional_tests/test_layout_and_styling.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        self.browser.find_element_by_id('id_sign_up').click()
+        signup_url = self.browser.current_url
+
+        # Login and Signup are on different pages
+        self.assertNotEqual(signup_url, login_url)
+
+        # the word "email" appears in the page
+        src = self.browser.page_source
+        text_found = re.search(r'password', src)
+        self.assertNotEqual(text_found, None)
```
Comment (justinanderson) [inline src/functional_tests/test_layout_and_styling.py]: Comment says 'email' but code says 'password'.

**Comment by justinanderson** [comment]
@RogerTangos Looks good other than the comments I left.

The bit about Vagrant will change soon. After the aws-migration branch is merged in, it'll be the recommended setup for development and local use. Though we'll need a headless webdriver for Selenium before the tests will work in a VM.

**Comment by RogerTangos** [comment]
updates as requested. Thanks for giving it a close look.

**Comment by RogerTangos** [comment]
yep. good to merge?

**Pull request merged**